lib/administration/ticket/include/irods/ticket_administration.hpp

```diff
@@ -0,0 +1,577 @@
+        /// \param[in] conn The communication object
+        /// \param[in] ticket_name The name of the ticket
+        /// \param[in] constraint The constraint that should be set on the ticket specified
+        ///
+        /// \since 4.3.1
+        template <typename TicketConstraint>
+        void set_ticket_constraint(RxComm& conn, const std::string_view ticket_name, const TicketConstraint& constraint)
+        {
+            if constexpr (std::is_same_v<use_count_constraint, TicketConstraint>) {
+                if (constraint.value < 0) {
+                    throw std::invalid_argument("Value of the n is invalid!");
+                }
+
+                detail::execute_ticket_operation(conn,
+                                                 "mod",
+                                                 ticket_name,
+                                                 "uses",
+                                                 std::to_string(constraint.value),
+                                                 "",
+                                                 "",
+                                                 false);
+            }
+            else if constexpr (std::is_same_v<n_writes_to_data_object_constraint, TicketConstraint>) {
+                if (constraint.value < 0) {
+                    throw std::invalid_argument("Value of the n is invalid!");
+                }
+
+                detail::execute_ticket_operation(conn,
+                                                 "mod",
+                                                 ticket_name,
+                                                 "write-file",
+                                                 std::to_string(constraint.value),
+                                                 "",
+                                                 "",
+                                                 false);
+            }
+            else if constexpr (std::is_same_v<n_write_bytes_constraint, TicketConstraint>) {
+                if (constraint.value < 0) {
+                    throw std::invalid_argument("Value of the n is invalid!");
+                }
+
+                detail::execute_ticket_operation(conn,
+                                                 "mod",
+                                                 ticket_name,
+                                                 "write-bytes",
+                                                 std::to_string(constraint.value),
+                                                 "",
+                                                 "",
+                                                 false);
+            }
+            else {
+                throw std::invalid_argument("Invalid constraint type.");
+            }
+        }
+
+        /// \brief Set constraint to the ticket specified
+        ///
+        /// \tparam TicketConstraint The template struct to indicate the type of constraint (only the ticket property constraint) 
```

Comment:
Add an empty line after this line so that the `\param` directive is more noticeable.